### ISSUE TITLE
Salmon Indexing Issue: Decoy has 1830 Seq but found none in refrence. 

### ISSUE THREAD

**Issue by abourke1** [COMBINE-lab/salmon#644]
Hey,

I'm pretty new to RNA-seq so hope this question is formatted okay! 

I've carried out my RNA-seq analysis using STAR into feature counts but wanted to redo the analysis using Salmon as everything I'm reading says its the way forward! I've tried to create a genome index using the following files.

``
wget http://ftp.ensembl.org/pub/release-103/fasta/cricetulus_griseus_picr/cdna/Cricetulus_griseus_picr.CriGri-PICR.cdna.all.fa.gz
wget http://ftp.ensembl.org/pub/release-103/fasta/cricetulus_griseus_picr/dna/Cricetulus_griseus_picr.CriGri-PICR.dna.nonchromosomal.fa.gz
``
Then I created the decoy file and the combined file  them using the  following. 
`
grep "^>" <(zcat Cricetulus_griseus_picr.CriGri-PICR.dna.nonchromosomal.fa.gz) | cut -d " " -f 1 > decoys.txt
cat Cricetulus_griseus_picr.CriGri-PICR.cdna.all.fa.gz Cricetulus_griseus_picr.CriGri-PICR.dna.nonchromosomal.fa.gz > gentrome.fa.gz
`
And finally ran the index using 
`./salmon index -t /data/fcr18ab/Refrence/New_Refrence_Files/gentrome.fa.gz -d /data/fcr18ab/Refrence/New_Refrence_Files/decoys.txt -p 1 -i /fastdata/fcr18ab/Adrian_salmon_index/salmon_index `

The error I get is shown below in the file. It says it can't find any of the decoy sequences in the combined file, but I have searched through them and they are there?

[Super_Low.txt](https://github.com/COMBINE-lab/salmon/files/6254711/Super_Low.txt)

Sorry if this is a super simple issue. The cell line is a cancer cell which has not got defined chromosomes. I'm sure its me messing up somewhere but I can't figure out where!

**Comment by TiphaineCMartin** [comment]
Hi, 
I have same issue using Salmon 1.4.0.

* `grep "^>" <GRCh38.d1.vd1.fa | cut -d " " -f 1 > decoys.txt`
*  `sed -i.bak -e 's/>//g' decoys.txt`
* `cat GDC.GRCh38_gencode.v22.fasta.gz GRCh38.d1.vd1.fa.gz > GRCh38.d1.vd1_gencode.v22_gentrome.fa.gz`
* `salmon index -t GRCh38.d1.vd1_gencode.v22_gentrome.fa.gz -d  decoys.txt -i salmon_1.4.0/ -p 12 -k 31 --gencode`

[2021-04-07 12:31:14.685] [puff::index::jointLog] [warning] Removed 789 transcripts that were sequence duplicates of indexed transcripts.
[2021-04-07 12:31:14.685] [puff::index::jointLog] [warning] If you wish to retain duplicate transcripts, please use the `--keepDuplicates` flag
[2021-04-07 12:31:14.685] [puff::index::jointLog] [critical] The decoy file contained the names of 2779 decoy sequences, but 0 were matched by sequences in the reference file provided. To prevent unintentional errors downstream, please ensure that the decoy file exactly matches with the fasta file that is being indexed.
[2021-04-07 12:31:15.194] [puff::index::jointLog] [error] The fixFasta phase failed with exit code 1

**Comment by abourke1** [comment]
Hey,

So I solved the issue over the weekend. My error was caused by my decoy file having my chromosome names with > in front of them meaning it couldn't find them when creating the index.
`
Instead of for example 

> grep "^>" <(zcat GRCm38.primary_assembly.genome.fa.gz) | cut -d " " -f 1 > decoys.txt

> sed -i -e 's/>//g' decoys.txt'

`
I used the following instead. 
`

> grep "^>" <(gunzip -c GRCm38.primary_assembly.genome.fa.gz) | cut -d " " -f 1 > decoys.txt

> sed -i.bak -e 's/>//g' decoys.txt

`
I'm guessing this isn't your issue but just in case it helps! I would check your "gentrome" file and look if the decoy and names within your gentrome file match up properly. That's how I realized the > was disrupting my analysis!

Hope this helps. 

Best Regards,
Adrian
